opal/mca/common/ucx/common_ucx_wpool.c

```diff
@@ -55,7 +57,7 @@ static opal_common_ucx_winfo_t *_winfo_create(opal_common_ucx_wpool_t *wpool)
     if (opal_common_ucx_thread_enabled || wpool->dflt_winfo == NULL) {
         memset(&worker_params, 0, sizeof(worker_params));
         worker_params.field_mask = UCP_WORKER_PARAM_FIELD_THREAD_MODE;
-        worker_params.thread_mode = UCS_THREAD_MODE_SINGLE;
+        worker_params.thread_mode = ompi_mpi_thread_provided == MPI_THREAD_SINGLE ? UCS_THREAD_MODE_SINGLE : UCS_THREAD_MODE_SERIALIZED;
```

Comment:
if the user initialized MPI with MPI_THREAD_MULTIPLE, will this be passed correctly to the initialization of the UCX worker when creating a window? I am not sure this fix here correctly captures this (or fixes it), see https://github.com/open-mpi/ompi/issues/13347